app/controllers/DataSetController.scala

```diff
@@ -78,14 +78,15 @@ class DataSetController @Inject()(val messagesApi: MessagesApi) extends Controll
   def list = UserAwareAction.async { implicit request =>
     UsingFilters(
       Filter("isEditable", (value: Boolean, el: DataSet) =>
-        el.isEditableBy(request.identity) && value || !el.isEditableBy(request.identity) && !value),
+        for {isEditable <- el.isEditableBy(request.identity)} yield {isEditable && value || !isEditable && !value}),
       Filter("isActive", (value: Boolean, el: DataSet) =>
-        el.isActive == value)
+        Fox.successful(el.isActive == value))
     ) { filter =>
       DataSetDAO.findAll.flatMap {
         dataSets =>
           for {
-            js <- Fox.serialCombined(filter.applyOn(dataSets))(d => DataSet.dataSetPublicWrites(d, request.identity))
+            filtered <- filter.applyOn(dataSets)
```

Comment:
:heart:

Reply:
`filter.applyOn` is now also asynchronous so the result has to be unpacked, which is what this line does :)